enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -164,6 +188,63 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
 
     }
 
+    private String generateAndRetrieveJWTToken(String tokenSignature, JWTInfoDto jwtInfoDto)
+            throws APISecurityException {
+        log.debug("Inside generateAndRetrieveJWTToken");
+
+        String endUserToken = null;
+        boolean valid = false;
+        String jwtTokenCacheKey = jwtInfoDto.getApicontext().concat(":").concat(jwtInfoDto.getVersion()).concat(":")
+                .concat(tokenSignature);
+        JWTConfigurationDto jwtConfigurationDto = ConfigHolder.getInstance().getConfig().getJwtConfigurationDto();
+        // Get the jwt generator class (Default jwt generator class)
+        //Todo: load the class from the configuration
+        jwtGenerator = JWTUtil.getApiMgtGatewayJWTGenerator();
+        if (jwtGenerator != null) {
+            jwtGenerator.setJWTConfigurationDto(jwtConfigurationDto);
+            if (isGatewayTokenCacheEnabled) {
+                try {
+                    Object token = CacheProvider.getGatewayJWTTokenCache().get(jwtTokenCacheKey);
+                    if (token != null) {
+                        endUserToken = (String) token;
+                        String[] splitToken = ((String) token).split("\\.");
```

Comment:
Could use just endUserToken here.

Reply:
Kept the same logic as in the carbon-apimgt repo.